pkg/cmd/roachtest/tests/mixed_version_decommission.go

```diff
@@ -40,12 +40,13 @@ func runDecommissionMixedVersions(
 
 	h := newDecommTestHelper(t, c)
 
-	// The v20.2 CLI can only be run against servers running v20.2. For this
-	// reason, we grab a handle on a specific server slated for an upgrade.
 	pinnedUpgrade := h.getRandNode()
 	t.L().Printf("pinned n%d for upgrade", pinnedUpgrade)
 
-	const suspectDuration = 5 * time.Second
+	// NB: The suspect duration must be at least 10s, as versions 23.2 and
+	// beyond will reset to the default of 30s if it fails validation, even if
+	// set by a previous version.
+	const suspectDuration = 10 * time.Second
```

Comment:
nit: You can use the constant `liveness.minTimeUntilNodeSuspect` - although you would have to make that a public const. It might be easier to see where this comes from and if it changes later.